ffi/memory.go

```diff
@@ -27,107 +27,116 @@ type KeyValue struct {
 	Value []byte
 }
 
-// extractErrorThenFree converts the cgo `Value` payload to either:
-// 1. a nil value, indicating no error, or
-// 2. a non-nil error, indicating an error occurred.
-// This should only be called when the `Value` is expected to only contain an error.
-// Otherwise, an error is returned.
-func extractErrorThenFree(v *C.struct_Value) error {
+// hashAndIDFromValue converts the cgo `Value` payload into:
+//
+//	case | data    | len   | meaning
+//
+// 1.    | nil     | 0     | invalid
+// 2.    | nil     | non-0 | proposal deleted everything
```

Comment:
to make sure I understand - the proposal ID is non-zero because it's a real proposal, but the root is empty because the trie ends up being empty as well?